raiden/transfer/state.py

```diff
@@ -596,8 +595,23 @@ def addresses_to_channel(
 def get_address_metadata(
     address: Address, route_states: List[RouteState]
 ) -> Optional[AddressMetadata]:
+
     for route_state in route_states:
         recipient_metadata = route_state.address_to_metadata.get(address, None)
         if recipient_metadata is not None:
             return recipient_metadata
     return None
+
+
+def fetch_address_metadata(
+    address: Address, route_states: Optional[List[RouteState]] = None
```

Comment:
Is there a need to have route_states here? I think this was old code from your previous draft where it would make sense to have it, IIRC.

Reply:
There's no need, that's why it's optional. But in case there is, we first check if by any chance we already have this metadata before making an external request. I think this extra check makes sense to try to avoid an api call.

Reply:
Ok then, I removed it ✅